src/Microsoft.VisualStudio.Threading.Analyzers.Tests/VSTHRD201CancelAfterSwitchToMainThreadAnalyzerTests.cs

```diff
@@ -0,0 +1,226 @@
+﻿namespace Microsoft.VisualStudio.Threading.Analyzers.Tests
+{
+    using System.Threading.Tasks;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.Testing;
+    using Xunit;
+    using Verify = CSharpCodeFixVerifier<VSTHRD201CancelAfterSwitchToMainThreadAnalyzer, VSTHRD201CancelAfterSwitchToMainThreadCodeFix>;
+
+    public class VSTHRD201CancelAfterSwitchToMainThreadAnalyzerTests
+    {
+        [Fact]
+        public async Task SwitchWithoutCheck_ProducesDiagnostic()
+        {
+            var test = @"
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Threading;
+
+class Test {
+    JoinableTaskFactory jtf;
+    async Task Foo(CancellationToken cancellationToken) {
+        await jtf.SwitchToMainThreadAsync(cancellationToken);
```

Comment:
Thanks. Is there any reason that would only work sometimes? I've tried it for some tests, and it doesn't seem to work, but in other places it does. I'll be happy to try the change here.

Reply:
There are some cases where the analyzer is not compatible with it.

* If the analyzer has more than one supported diagnostic, you can use `{|SpecificID:code|}` instead of `[|code|]` to indicate the specific diagnostic ID that is reported at that location. This works for C# compiler errors as well.
* If the analyzer has more than one supported diagnostic descriptor _with the same ID_, you cannot use markup to mark the locations because it can't tell which one is correct.

Reply:
OK, that's a lot to keep in mind, but it probably explains why I saw "inconsistent" results. Thanks.

When I tried applying this change, and changed the `WithSpan` line to:

```cs
            var expected = Verify.Diagnostic();
```

The test failed with:

```
Message: Mismatch between number of diagnostics returned, expected "2" actual "1"

Diagnostics:
// Test0.cs(9,15): warning VSTHRD201: Check for cancellation after calling SwitchToMainThreadAsync(CancellationToken).
GetCSharpResultAt(9, 15, VSTHRD201CancelAfterSwitchToMainThreadAnalyzer.VSTHRD201)


Expected: True
Actual:   False
```

What did I do wrong?

Reply:
> What did I do wrong?

Remove `var expected` and write the call like you normally would for a test with no diagnostics.